department-of-veterans-affairs/vets-website
# [Development[ HLR: Connect frontend to backend endpoints

## Description

This PR accomplishes:

- Get contestable issues endpoint
  - Shows an error message if the endpoint isn't available
  - Shows a message if the user has no contestable issues to proceed with
  - Shows the save-in-progress component when the user can proceed
  - Added unit tests
- Connect to submission endpoint
  - Added a submit transformer.
  - Note the [LightHouse](https://dev-developer.va.gov/explore/appeals/docs/decision_reviews?version=current)  contestable issues API provides a list of issues with differently named keys then what is expected in the posted LightHouse submitted contestable issues. 
  - Added unit tests with a maximal JSON & transformed maximal JSON mock data
- Updated code to use the latest LightHouse API contestable issues endpoint
- Other misc fixes
  - Fix JS error & indexed fragments issue
  - Remove out-of-date code & comments
  - Removes `changeCase` code which was not optimized to capitalize hyphenated (`Smith-Brown`) or apostrophed (`O'Donald`) names

## Testing done

Added & updated unit tests

## Screenshots

<details><summary>Connection error screenshot</summary>


![](https://user-images.githubusercontent.com/136959/86028948-2c087080-b9f8-11ea-8294-04220ee514ed.png)</details>

<details>
<summary>No contestable issues message alert</summary>


![](https://user-images.githubusercontent.com/136959/86029137-683bd100-b9f8-11ea-9280-6ab12a1a6ef1.png)</details>

<details><summary>Logged in user is able to proceed</summary>


![](https://user-images.githubusercontent.com/136959/86060460-596e1200-ba2a-11ea-836f-3b96d70bcc10.png)</details>

Related:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/1933
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/10713
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/9817

## Acceptance criteria

- [ ] HLR connected to contestable issues endpoint
- [ ] HLR connected to submission endpoint
- [ ] Contestable issues error message shown
- [ ] No contestable issues message shown (without a start the form button)

## Definition of done
- [x] Events are logged appropriately (BE error logging will be used for these endpoints)
